tests/testutils/aci-server/server.go

```diff
@@ -380,13 +380,28 @@ func newLocalListener(port PortType, protocol ProtocolType) net.Listener {
 			portNumber = 443
 		}
 	}
-	l, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", portNumber))
+	addrs, err := net.LookupHost("localhost")
 	if err != nil {
-		if l, err = net.Listen("tcp6", fmt.Sprintf("[::1]:%d", portNumber)); err != nil {
-			panic(fmt.Sprintf("aci test server: failed to listen on a port: %v", err))
+		panic(`aci test server: failed to look up "localhost", really`)
+	}
+	var lookupErrs []string
+	for try := range []int{0, 0} {
+		for _, addr := range addrs {
+			addrport := fmt.Sprintf("%s:%d", addr, portNumber)
+			l, err := net.Listen("tcp", addrport)
```

Comment:
No, it assumes either ipv4 or ipv6. `net.Listen("tcp4", …)` would force ipv4. See https://golang.org/pkg/net/#Dial for details.